hpi-archetype/pom.xml

```diff
@@ -15,15 +14,34 @@
   <!-- $ --><version>1.0-SNAPSHOT</version><!-- /$ -->
   <packaging>hpi</packaging>
 
+  <properties>
+    <!-- Baseline Jenkins version you use to build and test the plugin. Users must have this version or newer to run. -->
+    <jenkins.version>1.580.1</jenkins.version>
+    <!-- Java Level to use. Default is 7, but was only required by Jenkins >= 1.614. -->
+    <java.level>6</java.level>
+    <!-- Other properties you may want to use:
+         ~ jenkins-test-harness.version: The JTH version used to test plugin. The default is ${jenkins.version}.
+         ~ hpi-plugin.version: The HPI Maven Plugin version used by the plugin. Current default is 1.115.
+         ~ stapler-plugin.version: The Stapler Maven plugin version required by the plugin. Current default is 1.17
+         ~ java.level.test: The Java version to use to build the plugin tests. Current default is 7.
```

Comment:
By "not encouraging" you mean not "advertising" it in the comments, don't you? The switch could be kept as it seems to be useful for some plugins.